dts/arm/nuvoton/m46x.dtsi

```diff
@@ -0,0 +1,590 @@
+/*
+ * Copyright (c) 2023 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv7-m.dtsi>
+#include <mem.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include "m46x/sys.h"
+#include "m46x/sys_reg.h"
+#include "m46x/clk.h"
+#include "m46x/clk_reg.h"
+
+/ {
+	chosen {
+		zephyr,flash-controller = &fmc;
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m4f";
+			reg = <0>;
+		};
+	};
+
+	sram0: memory@20000000 {
+		compatible = "mmio-sram";
+		reg = <0x20000000 DT_SIZE_K(512)>;
+	};
+
+	sysclk: system-clock {
+		compatible = "fixed-clock";
+		clock-frequency = <200000000>;
+		#clock-cells = <0>;
+	};
+
+	soc {
+		scc: system-clock-controller@40000200 {
+			compatible = "nuvoton,numaker-scc";
+			reg = <0x40000200 0x100>;
+			#clock-cells = <0>;
+			/* hxt = "enable"; */
+			/* lxt = "enable"; */
+			clk-pclkdiv = <(CLK_PCLKDIV_APB0DIV_DIV2 | CLK_PCLKDIV_APB1DIV_DIV2)>;
+			core-clock = <200000000>;
+
+			pcc: peripheral-clock-controller {
+				compatible = "nuvoton,numaker-pcc";
+				#clock-cells = <3>;
+			};
+		};
+
+		fmc: flash-controller@4000c000 {
+			compatible = "nuvoton,numaker-fmc";
+			reg = <0x4000c000 0x110>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0: flash@0 {
+				compatible = "soc-nv-flash";
+				reg = <0 DT_SIZE_K(1024)>;
+				erase-block-size = <4096>;
+				write-block-size = <4>;
+			};
+		};
+
+		uart0: serial@40070000 {
+			compatible = "nuvoton,numaker-uart";
+			reg = <0x40070000 0x1000>;
+			interrupts = <36 0>;
+			reset = <UART0_RST>;
```

Comment:
where is the reset controller here?

Reply:
In `uart_numaker_init()`, call `SYS_ResetModule(config->id_rst)` to ensure H/W status clean.

Reply:
this needs a reset driver

Reply:
ST/GD and other platforms are as simple as that, but for consistency we should be using reset controller API.

Reply:
I see, most MPU chip with asert/deassert feature will need Linux reset driver.
I will check the capibility with only toggle function, how to fulfill.

Reply:
After check, it's doable and is great to consistency.
For simplified this PR, could I add reset driver in next PR ?

Reply:
Hi @gmarull,
Sorry for late reply due to consecutive holidays, 
I added the reset driver at commit [c92783d](https://github.com/zephyrproject-rtos/zephyr/pull/55406/commits/c92783ddb9b3f91afcdebf594b92209dab83dde0) in this PR.